### PR TITLE
Migrate User.GetProfilesInChannelByStatus to Sync by default #11461

### LINKED ISSUE #11461: Migrate "User.GetProfilesInChannelByStatus" to Sync by default
If you're interested please comment here and come [join our "Contributors" community channel](https://community.mattermost.com/core/channels/tickets) on our daily build server, where you can discuss questions with community members and the Mattermost core team. For technical advice or questions, please  [join our "Developers" community channel](https://community.mattermost.com/core/channels/developers).

New contributors please see our [Developer's Guide](https://developers.mattermost.com/contribute/getting-started/).

----

**Notes**: [Jira ticket](https://mattermost.atlassian.net/browse/MM-16659)

Mattermost is migrating its Store layer to be sync by default, and only use Async when needed and we're looking for contributors to help with that effort. This Help Wanted issue is to migrate the `GetProfilesInChannelByStatus` in the `User` store.

The expected way to implement it is, go to the `User` store implementation in the `store/sqlstore/` directory, modify the method `GetProfilesInChannelByStatus` to return directly an object from the `model` module, and a `*model.AppError` (removing the `store.Do` wrapper). After that, you must modify the interface defined in `store/store.go` to match with the changes. Then, you should execute `make store-mocks` to rebuild the mocks with the new interface. And finally, modify the rest of the code (tests included) to use the new interface of the function properly.

Example: mattermost/mattermost-server#10613

### PR DESCRIPTION
**Pull request by Inconnu08**
#### Summary
Migrate `User.GetProfilesInChannelByStatus` to sync by default.

#### Ticket Link
Fixes #11461

### CODE REVIEW

File: app/user.go [hunk 1]
```diff
@@ -505,11 +505,11 @@ func (a *App) GetUsersInChannel(channelId string, offset int, limit int) ([]*mod
 }
 
 func (a *App) GetUsersInChannelByStatus(channelId string, offset int, limit int) ([]*model.User, *model.AppError) {
-	result := <-a.Srv.Store.User().GetProfilesInChannelByStatus(channelId, offset, limit)
-	if result.Err != nil {
-		return nil, result.Err
+	result, err := a.Srv.Store.User().GetProfilesInChannelByStatus(channelId, offset, limit)
```
Comment (jespino) [inline app/user.go]: You can simplify it using `return a.Srv.Store.User().GetProfilesInChannelByStatus(channelId, offset, limit)` directly.

File: store/sqlstore/user_store.go [hunk 2]
```diff
@@ -557,41 +557,37 @@ func (us SqlUserStore) GetProfilesInChannel(channelId string, offset int, limit
...
-			LeftJoin("Status s ON ( s.UserId = u.Id )").
-			Where("cm.ChannelId = ?", channelId).
-			OrderBy(`
+func (us SqlUserStore) GetProfilesInChannelByStatus(channelId string, offset int, limit int) ([]*model.User, *model.AppError) {
+	query := us.usersQuery.
+		Join("ChannelMembers cm ON ( cm.UserId = u.Id )").
+		LeftJoin("Status s ON ( s.UserId = u.Id )").
+		Where("cm.ChannelId = ?", channelId).
+		OrderBy(`
 				CASE s.Status
```
Comment (jespino) [inline store/sqlstore/user_store.go]: Would be nice to remove one tab from all the lines of this string to maintain the consistency in the indentation.

File: store/storetest/user_store.go [hunk 3]
```diff
@@ -840,15 +840,15 @@ func testUserStoreGetProfilesInChannelByStatus(t *testing.T, ss store.Store) {
...
+		result, err := ss.User().GetProfilesInChannelByStatus(c1.Id, 0, 100)
+		require.Nil(t, err)
+		assert.Equal(t, []*model.User{sanitized(u3), sanitized(u2), sanitized(u1)}, result)
 	})
 
 	t.Run("get in channel 2 by status, offset 0, limit 1", func(t *testing.T) {
-		result := <-ss.User().GetProfilesInChannelByStatus(c2.Id, 0, 1)
-		require.Nil(t, result.Err)
-		assert.Equal(t, []*model.User{sanitized(u1)}, result.Data.([]*model.User))
+		result, err := ss.User().GetProfilesInChannelByStatus(c2.Id, 0, 1)
```
Comment (jespino) [inline store/storetest/user_store.go]: Probably we can change here `result` with `users` which is more explicit.

File: store/storetest/user_store.go [hunk 4]
```diff
@@ -840,15 +840,15 @@ func testUserStoreGetProfilesInChannelByStatus(t *testing.T, ss store.Store) {
 	}))
 
 	t.Run("get in channel 1 by status, offset 0, limit 100", func(t *testing.T) {
-		result := <-ss.User().GetProfilesInChannelByStatus(c1.Id, 0, 100)
-		require.Nil(t, result.Err)
-		assert.Equal(t, []*model.User{sanitized(u3), sanitized(u2), sanitized(u1)}, result.Data.([]*model.User))
+		result, err := ss.User().GetProfilesInChannelByStatus(c1.Id, 0, 100)
```
Comment (jespino) [inline store/storetest/user_store.go]: Probably we can change here `result` with `users` which is more explicit.

**Comment by Inconnu08** [comment]
@jespino done

File: store/storetest/user_store.go [hunk 5]
```diff
@@ -840,15 +840,15 @@ func testUserStoreGetProfilesInChannelByStatus(t *testing.T, ss store.Store) {
 	}))
 
 	t.Run("get in channel 1 by status, offset 0, limit 100", func(t *testing.T) {
-		result, err := ss.User().GetProfilesInChannelByStatus(c1.Id, 0, 100)
+		user, err := ss.User().GetProfilesInChannelByStatus(c1.Id, 0, 100)
```
Comment (jespino) [inline store/storetest/user_store.go]: `users` in plural.

Comment (Inconnu08) [reply store/storetest/user_store.go]: @jespino done

File: store/storetest/user_store.go [hunk 6]
```diff
@@ -840,15 +840,15 @@ func testUserStoreGetProfilesInChannelByStatus(t *testing.T, ss store.Store) {
...
+		users, err := ss.User().GetProfilesInChannelByStatus(c1.Id, 0, 100)
+		require.Nil(t, err)
+		assert.Equal(t, []*model.User{sanitized(u3), sanitized(u2), sanitized(u1)}, users)
 	})
 
 	t.Run("get in channel 2 by status, offset 0, limit 1", func(t *testing.T) {
-		result := <-ss.User().GetProfilesInChannelByStatus(c2.Id, 0, 1)
-		require.Nil(t, result.Err)
-		assert.Equal(t, []*model.User{sanitized(u1)}, result.Data.([]*model.User))
+		user, err := ss.User().GetProfilesInChannelByStatus(c2.Id, 0, 1)
```
Comment (jespino) [inline store/storetest/user_store.go]: And same heres, replace `user` with `users` please

**Comment by Inconnu08** [comment]
@jespino and it's done

**Comment by Inconnu08** [comment]
Awesome! 🎉

**Pull request merged**